src/soundanalysis.cs

```diff
@@ -153,4 +183,31 @@ interface SNResultsObserving {
 		[Export ("requestDidComplete:")]
 		void DidComplete (ISNRequest request);
 	}
+
+	[iOS (15,0), Mac (12,0), Watch (8,0), TV (15,0), MacCatalyst (15,0)]
+	enum SNClassifierIdentifier {
+		[Field ("SNClassifierIdentifierVersion1")]
+		Version1,
+	}
+
+	[Watch (8,0), TV (15,0), Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface SNTimeDurationConstraint /* privately conforms to NSCoding, NSCopying, and NSSecureCoding */
+	{
+		[Export ("initWithEnumeratedDurations:")]
+		IntPtr Constructor (NSValue[] enumeratedDurations);
+
+		[Export ("initWithDurationRange:")]
+		IntPtr Constructor (CMTimeRange durationRange);
+
+		[Export ("type", ArgumentSemantic.Assign)]
+		SNTimeDurationConstraintType Type { get; }
+
+		[Export ("enumeratedDurations", ArgumentSemantic.Strong)]
+		NSValue[] EnumeratedDurations { get; }
```

Comment:
The `enumeratedDurations` property is defined to be "an array of CMTime structures (boxed in NSValue instances)", so you can use `BindAs` here to expose it as an array of CMTime.

Reply:
I tried this and showed @dalexsoto and he came to the conclusion that CMTime arrays are not supported yet

Reply:
We have tests for BindAs with CMTime arrays:

https://github.com/xamarin/xamarin-macios/blob/ffc17b6ce287620b7cd5785524ebf607d69b09c1/tests/generator/bindastests.cs#L102-L104

Reply:
I'll give it another shot!

Reply:
So after making this change: https://github.com/xamarin/xamarin-macios/pull/12181/files#diff-771cfc4720d50e07d24a25397a8721722a43b16db581b2cb4259de08f0a61414L15

Adding these BindAs's gives me these errors: https://gist.github.com/tj-devel709/7a2427e85b5fdd310505638decd8b34d
I am pretty confused on why it is trying to convert from CoreMedia.CMTime[] to Foundation.NSObject though

Reply:
Can you gist `build/watch/watch/SoundAnalysis/SNTimeDurationConstraint.g.cs` too?

Reply:
https://gist.github.com/tj-devel709/dfea500a2c3bfa0c316523e7ce2b74b6

Reply:
I believe that's because `CMTime` didn't used to be in watchOS, so we defined it to be `NSObject`, like this:

https://github.com/xamarin/xamarin-macios/blob/478b58897fc0b30920ac225750c1d4d4f20343fd/src/foundation.cs#L88

That wasn't a problem while `CMTime` wasn't actually used in any watchOS API, but now that it is (and also because `CMTime` was introduced to watchOS in watchOS 6.0), the fact that we define it to be `NSObject` is causing these weird errors.

The fix is to expose `CMTime` in watchOS as well, and remove these hacks defining `CMTime` as `NSObject` - and the same is true for `CMTimeMapping` and `CMTimeRange` just below:

https://github.com/xamarin/xamarin-macios/blob/478b58897fc0b30920ac225750c1d4d4f20343fd/src/foundation.cs#L89-L90

If would probably be best to do that in a different PR, and once that PR is merged into main, then merge main into this PR, and the BindAs attribute change should compile just fine.

Reply:
Thanks for the rundown! I will create that PR now!

Reply:
Okay, PR is now created: https://github.com/xamarin/xamarin-macios/pull/12280

Reply:
I have fixed this issue in the last push.